### PR TITLE
Dual variables for `PowConeND`

### PR DESCRIPTION
**Pull request by aryamanjeendgar**
## Description
This PR implements (long-coming \:)) dual variable support for `PowConeND`.

The major logic of the implementation is couched inside of `invert::Exotic2Common::exotic2common.py` --- in addition, keeping in line with my on-going contributions for building a user-level infrastructure around optimality condition verification in CVXPY, I have also implemented the `_dual_cone` method for `PowConeND` (so now, one can call `dual_residual` and verify dual-feasibility for power cone programs).

The tests can be found inside of `test_KKT.py` (I decided to use the existing KKT infrastructure for verifying the correctness of my implementation)

 Issue link (if applicable): #1222 

## Type of change
- [x] New feature (backwards compatible)
- [] New feature (breaking API changes)
- [] Bug fix
- [] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by aryamanjeendgar** [comment]
Tagging @rileyjmurray @SteveDiamond @phschiele for review and suggestions, thanks!

File: cvxpy/reductions/cone2cone/exotic2common.py [hunk 1]
```diff
@@ -91,3 +92,31 @@ class Exotic2Common(Canonicalization):
...
+        pvars = {vid: solution.primal_vars[vid] for vid in inverse_data.id_map
+                 if vid in solution.primal_vars}
+        dvars = {orig_id: solution.dual_vars[vid]
+                 for orig_id, vid in inverse_data.cons_id_map.items()
+                 if vid in solution.dual_vars}
+
+        dv = {}
+        for cons_id, cons in inverse_data.id2cons.items():
+            if isinstance(cons, PowConeND):
+                idx = int(f'{cons_id}')
```
Comment (PTNobel) [inline cvxpy/reductions/cone2cone/exotic2common.py:106]: What is going on here? This is strange notation. Why turn it into a string?

File: cvxpy/reductions/cone2cone/exotic2common.py [hunk 2]
```diff
@@ -91,3 +92,31 @@ class Exotic2Common(Canonicalization):
...
+                div_size = int(dvars[idx].shape[1] / cons.args[1].shape[0])
```
Comment (PTNobel) [inline cvxpy/reductions/cone2cone/exotic2common.py:107]: Should we use integer division `\\` instead? This risks weird truncation issues with floating points.

**Review by PTNobel — CHANGES_REQUESTED** [review]
I skipped reviewing the tests and didn't check the math, but had a few comments on the code style.

File: cvxpy/constraints/power.py [hunk 3]
```diff
@@ -273,5 +273,32 @@ def is_dqcp(self) -> bool:
...
+        """Implements the dual cone of PowConeND See Pg 85
+        of the MOSEK modelling cookbook for more information"""
+        if args is None or args == ():
+            scaled_duals = self.dual_variables[0]/self.alpha
+            return PowConeND(scaled_duals, self.dual_variables[1], self.alpha, axis=self.axis)
+        else:
+            # some assertions for verifying `args`
+            def f(x):
+                return x.shape
+            args_shapes = list(map(f, args))
```
[the remark this answers is not in the document]
Comment (rileyjmurray) [reply cvxpy/constraints/power.py:299]: Agreed

File: cvxpy/reductions/cone2cone/exotic2common.py (hunk 1 above)
Comment (rileyjmurray) [reply cvxpy/reductions/cone2cone/exotic2common.py:106]: Good question. @aryamanjeendgar?

File: cvxpy/reductions/cone2cone/exotic2common.py (hunk 2 above)
Comment (rileyjmurray) [reply cvxpy/reductions/cone2cone/exotic2common.py:107]: I agree that integer division would be preferable (but it's ``//``, not ``\\``).

File: cvxpy/constraints/power.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rileyjmurray) [reply cvxpy/constraints/power.py:299]: Yes, I'd make the change to list comprehensions in the other classes as well. Thanks for checking!

**Review by rileyjmurray — APPROVED** [review]
Looks good to me!

**Pull request merged**